aiohomekit/controller/ble/pairing.py

```diff
@@ -802,6 +802,43 @@ async def put_characteristics(
 
         return results
 
+    async def thread_provision(self, network_name: str, channel: int, pan_id: str, extended_pan_id: str, network_key: str, unknown: int) -> None:
+        """Provision a device with Thread network credentials."""
+        thread_service = self.accessories.aid(1).services.first(service_type=ServicesTypes.THREAD_TRANSPORT)
+        thread_control = thread_service.characteristics.first(char_types=[CharacteristicsTypes.THREAD_CONTROL_POINT])
+
+        thread_tlv = TLV.encode_list([
+            (1, bytes(network_name, encoding="ascii")),
+            (2, channel.to_bytes(1, byteorder="little")),
+            (3, int(pan_id, base=16).to_bytes(2, byteorder="little")),
+            (4, bytes.fromhex(extended_pan_id)),
+            (5, bytes.fromhex(network_key)),
+        ])
+        request_tlv = TLV.encode_list(
+            # TLV 1 is some sort of write/provision OpCode
+            # TLV 2 contains the Thread network details
+            # TLV 3 seems to be a bitfield or identifier; iOS sends 0 & Android sends 1
+            [(1, b"\x01"), (2, thread_tlv), (3, unknown.to_bytes(1, byteorder="little"))]
+        )
+        request_tlv = TLV.encode_list(
+            [
+                (TLV.kTLVHAPParamParamReturnResponse, bytearray(b"\x01")),
+                (TLV.kTLVHAPParamValue, request_tlv),
+            ]
+        )
+
+        try:
+            await self._async_request(OpCode.CHAR_WRITE, thread_control, request_tlv)
```

Comment:
I'm about to change the return type from async request to be a tlv since we need to support fragmented tics

Reply:
Actually it looks like the tlv fragmentation on top of the pdu fragmentation can only happen during pairing so I won't need to change it

Work in progress
https://github.com/Jc2k/aiohomekit/pull/149